src/client/clientmap.h

```diff
@@ -119,10 +119,15 @@ class ClientMap : public Map, public scene::ISceneNode
 	}
 
 	void getBlocksInViewRange(v3s16 cam_pos_nodes,
-		v3s16 *p_blocks_min, v3s16 *p_blocks_max);
+		v3s16 *p_blocks_min, v3s16 *p_blocks_max, float range=-1.0f);
 	void updateDrawList();
+	void updateDrawListShadow(
+			v3f shadow_light_pos, v3f shadow_light_dir, float shadow_range);
```

Comment:
these v3f should be passed as const references. This prevent also a useless v3f copy at the beginning of the function

Reply:
I think passing small vectors in registers is <i>less</i> efficient, a `v3f` is 12 bytes and easily fits in registers or on the stack.

Reply:
"const v3f & " is the way to go here.

Reply:
Why?

Reply:
Because const correctness, the reference is 8 bytes(4 in x32) instead of 12+ of v3f.
And const to ensure at compile time that the value of that parameter isn't modified.